core/server/common/src/main/java/alluxio/master/journal/JournalEntryAssociation.java

```diff
@@ -52,7 +52,8 @@ public static String getMasterForEntry(JournalEntry entry) {
         || entry.hasBlockInfo()) {
       return Constants.BLOCK_MASTER_NAME;
     }
-    if (entry.hasPathProperties()
+    if (entry.hasClusterInfo()
```

Comment:
it seems 2.0.1 journal will break journal backwards compatibility by throwing `IllegalStateException`